source/SpinalHDL/Data types/bool.rst

```diff
@@ -9,7 +9,52 @@ Bool
 Description
 ^^^^^^^^^^^
 
-The ``Bool`` type corresponds to a boolean value (True or False).
+The ``Bool`` type corresponds to a boolean value (True or False) or a single bit/wire
+used in a hardware design.  While named similarly it should not be confused with
+Scala `Boolean` type which does not describe hardware but truth values in the Scala
+generator code.
+
+An important concept and rule-of-thumb to understand is that the Scala
+`Boolean` type is used in places where ahead-of-time HDL code-generation
+decision making is occuring in Scala code.  Like any regular program it affects
+execution of the Scala program that is SpinalHDL at the time the program is
+being run to perform HDL code generation.
+
+Therefore the value of a Scala `Boolean` can not be observed from hardware,
+because it only exists ahead-of-time in the SpinalHDL program at the time of
+HDL code-gen.
+
+In scenarios where you might need this for your design, for example to pass a
+value (that maybe acting as a parameterized constant input) from Scala into your
+hardware design, you can type convert it to Bool with the constructor `Bool(value: Boolean)`.
+
+Similarly the value of a SpinalHDL `Bool` can not be seen at code-generation, all
+that can be seen and manipulated is the HDL construct concerning a `wire` and how it
+is routed (through modules/Components), driven (sourced) and connected (sunk).
+
+The signal direction of assignment operators `:=` is managed by SpinalHDL.
+The use of the Bool instance on the left-hand-side or the right-hand-side of the
+assignment operator `:=` dictates if it is a source (provides state) or sink
+(captures state) for a given assignment.
+
+Multiple uses of the assignment operator are allowed, such that it is normal
+for a signal wire to act as a source (provides a value to drive HDL state) to be
+able to connect and drive multiple inputs of other HDL constructs.  When a Bool
+instance used as a source the order the assignment statments appear or are
+executed in Scala does not matter, unlike when it is used as a sink
+(captures state).
+
+When multiple assignment operators drive the Bool (the Bool is on the
```

Comment:
Only if all but one of the assignments is in a condition, see chapter on assignments.

Reply:
I think you mean the `when` condition and other similar HDL constructs.  I found this to be a given and it has not been an important detail to understand at any point so far.

That is: Yes if I wrap my assignment in a Scala or hardware condition it is going to modulate if and when that assignment occurs.  This is a given.

If there is a specific contrived example that can highlight a not-obvious SpinalHDL precedence rule that breaks the general simple rule(s), maybe there should be a section demonstrating it, but I would think it would be under intermediate/advanced user areas (end of chapters/erratum) and not something brought up when explaining initial concepts.

Kind of learn this first... now you have understand that... in this scenario those rules break.  This is the exception of the rule.